antarest/storage/service.py

```diff
@@ -182,22 +192,32 @@ def _save_metadata(
             metadata.groups = [group]
         self.repository.save(metadata)
 
-    def _get_metadata_according_permission(
-        self, user: Optional[User], uuid: str
-    ) -> Metadata:
-        if not user:
-            raise UserHasNotPermissionError()
+    def _get_metadata(self, uuid: str) -> Metadata:
 
         md = self.repository.get(uuid)
         if not md:
+            sanitized = StorageServiceUtils.sanitize(uuid)
             logger.warning(
                 f"Study %s not found in metadata db",
-                StorageServiceUtils.sanitize(uuid),
+                sanitized,
             )
+            raise StudyNotFoundError(uuid)
+        return md
+
+    def _assert_permission(
```

Comment:
// why not _check_permission like before ?

Reply:
When code could raise an error during check I prefere to use `assert` key word. However, if an object method begin by `assert_*` there are troubles when this object will be mocked. But here we can because we prefix with `_`

Reply:
"when code could raise an error", well it's pretty much everything.